### DISCUSSION TITLE
Wrench Cone Residual Foot Parameters

### DISCUSSION THREAD

**Discussion by mattephi** [Simple-Robotics/aligator#168] [Ideas]
As far as I know, [residual](https://simple-robotics.github.io/aligator/structaligator_1_1WrenchConeResidualTpl.html) has two parameters - length and width of the foot. Does it imply that the robot should have the joint at the middle of the foot? If it does, are there any workarounds you are aware of if the robot foot is connected with some translation rather than in the middle?

**Comment by ManifoldFR** [comment]
Hi,
I think this is just related to the contact surface itself which is modelled as a rectangle. IIRC it's just that the resulting equivalent wrench cone (in terms of contact forces) applies at the centre of foot. It's not really related to where the foot connects to the rest of the robot I think
Does that sound right @edantec ?

**Reply by mattephi** [reply]
So what do you think? We think the point of calculating the forces is pretty important, but  probably just misunderstand the formulas application, it would be nice if you share some insights.

**Reply by edantec** [reply]
The wrench cone formula used in this work is derived from the following paper: _Stability of Surface Contacts for Humanoid Robots:
Closed-Form Formulae of the Contact Wrench Cone for Rectangular Support Areas_ from Caron et al. The resulting wrench is computed at the center of the contact link, which is rectangular. The forces are then recast at the joint level using the relative placement of the contact point with respect to the joint.
